DependencyInjection/DoctrineExtension.php

```diff
@@ -793,4 +788,22 @@ public function getConfiguration(array $config, ContainerBuilder $container)
     {
         return new Configuration($container->getParameter('kernel.debug'));
     }
+
+    private function loadMessengerServices(ContainerBuilder $container) : void
+    {
+        // If the Messenger component is installed and the doctrine transaction middleware is available, wire it:
+        if (! interface_exists(MessageBusInterface::class) || ! class_exists(DoctrineTransactionMiddleware::class)) {
+            return;
+        }
+
+        $loader = new XmlFileLoader($container, new FileLocator(__DIR__ . '/../Resources/config'));
+        $loader->load('messenger.xml');
+
+        if (! class_exists(DoctrineTransportFactory::class)) {
```

Comment:
I was going to ask why load a file with a class that does not exist, but I guess it's in case other services with class that do exist are added later? Should the definition maybe be removed from the container if the class does not exist?

Reply:
The file was added previously. I thought it was best to add the service there as well but to leave it untagged. Is it necessary to delete the service? We can also make the service private and have it removed automatically during container compilation.

Reply:
> The file was added previously.

Oh right, didn't see that other service, my bad.

> Is it necessary to delete the service?

I don't think it could be harmful, my suggestion was maybe a bit paranoid :P

Reply:
There is no need to remove the service, it will be removed automatically by the container when not used.